ember-headless-form/src/components/-private/field.hbs

```diff
@@ -5,8 +5,14 @@
       input=(component
         (ensure-safe-component this.InputComponent)
         fieldId=uuid
-        value=this.value
-        setValue=setValue
+        value=this.valueAsString
+        setValue=this.setValue
+      )
+      checkbox=(component
+        (ensure-safe-component this.CheckboxComponent)
```

Comment:
Question: what does `ensure-safe-component` do?

Reply:
Under Embroider, the use of the `component` helper (which we need here for currying of arguments) will throw a build-time error if you pass a non-string as the first argument, as it cannot statically analyze that to ensure the requested component is pulled into the bundle. But I didn't want to make the yielded component resolvable through a global string (like `headless-form/control/checkbox`), as the component should only be used as a contextual component, and those global string lookups should go away anyway in the future with the newer template-imports.

`ensure-safe-component` is the Embroider-endorsed escape hatch here, which makes Embroider not fail the build, but it will still make sure that what you passed to it is already something that has been imported elsewhere, and as such is already part of the Embroider (strict-mode) bundle. For the detailed explanation of its use cases, see https://github.com/embroider-build/embroider/blob/main/REPLACING-COMPONENT-HELPER.md